### PR TITLE
Cancel test if disk not provided

### PR DESCRIPTION
**Pull request by AYUSHJAIN951**
self cancel test if disk is not provided in yaml file in rawread.py test.

Signed-off-by: Ayush Jain <ayush.jain3@amd.com>

### CODE REVIEW

**Comment by AYUSHJAIN951** [comment]
JOB ID     : 3895b7271281a8c866e6be7504f6ed9a38e00bdc
JOB LOG    : /root/tests/results/job-2023-01-24T13.42-3895b72/job.log
 (1/1) avocado-misc-tests/io/disk/rawread.py:Rawread.test;run-dfbc: STARTED
 (1/1) avocado-misc-tests/io/disk/rawread.py:Rawread.test;run-dfbc:  PASS (51.71 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/tests/results/job-2023-01-24T13.42-3895b72/results.html
JOB TIME   : 53.37 s

File: io/disk/rawread.py [hunk 1]
```diff
@@ -38,6 +38,11 @@ def setUp(self):
         checking install of required packages and extract and
         compile of rawread suit.
         """
+        self.disk = self.params.get("disk", default=None)
+
+        if not self.disk:
+            self.cancel("Missing disk %s in OS" % self.disk)
```
Comment (Naresh-ibm) [inline io/disk/rawread.py:44]: @AYUSHJAIN951 I think it is not issing in the OS rather user did not provied the disk.
So please change the message to like "User needs to provide the disk", or "disk not provided" or "please provide the disk to run the test". any similar message looks good.

Please change it accordingly and re-send the Patch.

**Review by Naresh-ibm — CHANGES_REQUESTED** [review]

**Comment by Naresh-ibm** [comment]
> JOB ID : 3895b7271281a8c866e6be7504f6ed9a38e00bdc JOB LOG : /root/tests/results/job-2023-01-24T13.42-3895b72/job.log (1/1) avocado-misc-tests/io/disk/rawread.py:Rawread.test;run-dfbc: STARTED (1/1) avocado-misc-tests/io/disk/rawread.py:Rawread.test;run-dfbc: PASS (51.71 s) RESULTS : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0 JOB HTML : /root/tests/results/job-2023-01-24T13.42-3895b72/results.html JOB TIME : 53.37 s

@AYUSHJAIN951 just a suggestion, you should also add the logs which cancel the test, as you are adding the code which cancels the test on disk is not given.

**Comment by AYUSHJAIN951** [comment]
JOB ID     : c11f4c03b39061e720f2872958700a5ae53930ce
JOB LOG    : /root/tests/results/job-2023-01-26T06.27-c11f4c0/job.log
 (1/1) rawread.py:Rawread.test: STARTED
 (1/1) rawread.py:Rawread.test: CANCEL: Please provide disk to run the test (0.64 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 1
JOB HTML   : /root/tests/results/job-2023-01-26T06.27-c11f4c0/results.html
JOB TIME   : 2.35 s

**Comment by AYUSHJAIN951** [comment]
when disk passed through yaml file
JOB ID     : 0bedfa3d45448411c7a602e5231c769359ff8074
JOB LOG    : /root/tests/results/job-2023-01-26T06.33-0bedfa3/job.log
 (1/1) rawread.py:Rawread.test;run-5955: STARTED
 (1/1) rawread.py:Rawread.test;run-5955: PASS (51.29 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/tests/results/job-2023-01-26T06.33-0bedfa3/results.html
JOB TIME   : 52.91 s

**Comment by AYUSHJAIN951** [comment]
Hi @Naresh-ibm  can you please review it again.

**Review by Naresh-ibm — APPROVED** [review]
@AYUSHJAIN951 Thanks for this patch.
LGTM

**Pull request merged**